src/endpoints/mex/mex.pair.service.ts

```diff
@@ -26,65 +26,60 @@ export class MexPairService {
   ) { }
 
   async refreshMexPairs(): Promise<void> {
-    const pairs = await this.getAllMexPairsRaw();
+    const pairs = await this.getAllMexPairsRaw(false);
     await this.cachingService.setRemote(CacheInfo.MexPairs.key, pairs, CacheInfo.MexPairs.ttl);
     await this.cachingService.setLocal(CacheInfo.MexPairs.key, pairs, Constants.oneSecond() * 30);
   }
 
   async getMexPairs(from: number, size: number, filter?: MexPairsFilter): Promise<any> {
-    let allMexPairs = await this.getAllMexPairs();
+    let allMexPairs = await this.getAllMexPairs(filter?.includeFarms ?? false);
     allMexPairs = this.applyFilters(allMexPairs, filter);
 
     return allMexPairs.slice(from, from + size);
   }
 
-
-  async getMexPair(baseId: string, quoteId: string): Promise<MexPair | undefined> {
-    const allMexPairs = await this.getAllMexPairs();
+  async getMexPair(baseId: string, quoteId: string, includeFarms: boolean = false): Promise<MexPair | undefined> {
+    const allMexPairs = await this.getAllMexPairs(includeFarms);
     return allMexPairs.find(pair => pair.baseId === baseId && pair.quoteId === quoteId);
   }
 
-  async getAllMexPairs(): Promise<MexPair[]> {
+  async getAllMexPairs(includeFarms: boolean = false): Promise<MexPair[]> {
     if (!this.apiConfigService.isExchangeEnabled()) {
       return [];
     }
 
+    const cacheKey = includeFarms ? CacheInfo.MexPairsWithFarms.key : CacheInfo.MexPairs.key;
+    const ttl = includeFarms ? CacheInfo.MexPairsWithFarms.ttl : CacheInfo.MexPairs.ttl;
+
     return await this.cachingService.getOrSet(
-      CacheInfo.MexPairs.key,
-      async () => await this.getAllMexPairsRaw(),
-      CacheInfo.MexPairs.ttl,
+      cacheKey,
+      async () => await this.getAllMexPairsRaw(includeFarms),
+      ttl,
       Constants.oneSecond() * 30,
     );
   }
 
   async getMexPairsCount(filter?: MexPairsFilter): Promise<number> {
-    const mexPairs = await this.getAllMexPairs();
+    const mexPairs = await this.getAllMexPairs(filter?.includeFarms ?? false);
     const filteredPairs = this.applyFilters(mexPairs, filter);
 
     return filteredPairs.length;
   }
 
-  async getAllMexPairsRaw(): Promise<MexPair[]> {
+  async getAllMexPairsRaw(includeFarms: boolean = false): Promise<MexPair[]> {
```

Comment:
should we create a new class `MexPairsOptions` or `MexPairsFilter` that will contain the `includeFarms` field? Passing a boolean to a function is not the best practice